### PR TITLE
Rename bamshuf to collate

### PR DESCRIPTION
**Pull request by mp15**
Now that bamshuf can handle CRAM files we have renamed it so that it has a more appropriate name.

### CODE REVIEW

**Comment by jmarshall** [comment]
I thought _collate_ meant this too, but AFAICT it is actually about [sort order](https://en.wikipedia.org/wiki/Collation).  SQL uses `group by`, of course.  Do we have evidence that this usage of collate is common outside our (samtools-core-, Sanger-, or UK-) enclave?

**Comment by jmarshall** [comment]
Documentation or otherwise needs to make clear that this is doing some nice randomisation as well as just plain collation.

**Comment by daviesrob** [comment]
I discovered the other day that GNU sort does something similar to this, which it calls 'random-sort'.  This is what the info page says about it:

```
`-R'
`--random-sort'
`--sort=random'
     Sort by hashing the input keys and then sorting the hash values.
     Choose the hash function at random, ensuring that it is free of
     collisions so that differing keys have differing hash values.
     This is like a random permutation of the inputs (*note shuf
     invocation::), except that keys with the same value sort together.
```

File: bamshuf.c [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/*  bamshuf.c -- bamshuf subcommand.
```
Comment (mp15) [inline bamshuf.c]: Oops shouldn't change file name here

**Comment by jmarshall** [comment]
Merged with trivial doco changes as 97c9883e139a127a8b36353b0e7ec12c8319ef38.

**Pull request closed without merging**